hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java

```diff
@@ -53,41 +62,100 @@ public ScheduledTaskDescriptor() {
     public ScheduledTaskDescriptor(TaskDefinition definition) {
         this.definition = definition;
         this.state = new HashMap();
-        this.stats.set(new ScheduledTaskStatisticsImpl());
+        this.stats = new ScheduledTaskStatisticsImpl();
     }
+
     public ScheduledTaskDescriptor(TaskDefinition definition,
-                                   Map<?, ?> state, ScheduledTaskStatisticsImpl stats) {
+                                   Map<?, ?> state, ScheduledTaskStatisticsImpl stats,
+                                   ScheduledTaskResult result) {
         this.definition = definition;
-        this.stats.set(stats);
+        this.stats = stats;
         this.state = state;
+        this.resultRef.set(result);
     }
 
     public TaskDefinition getDefinition() {
         return definition;
     }
 
     ScheduledTaskStatisticsImpl getStatsSnapshot() {
-        return stats.get().snapshot();
+        return stats.snapshot();
     }
 
-    void setStats(ScheduledTaskStatisticsImpl stats) {
-        this.stats.set(stats);
+    Map<?, ?> getStateSnapshot() {
```

Comment:
I guess this can be renamed to `getState` since we are no longer creating a new map.